
--- a/Makefile	2021-09-13 12:56:16.437572538 +0200
+++ b/Makefile	2021-09-13 12:55:45.060952535 +0200
@@ -233,6 +233,16 @@
    endif
    
    
+   # Rockchip rk356x
+   ifneq (,$(findstring rk356x,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
+      HAVE_NEON = 1
+      WITH_DYNAREC=aarch64
+      CPUFLAGS += -D__NEON_OPT -march=armv8.2-a -mcpu=cortex-a55 -mtune=cortex-a55
+   endif
+
    # Classic Platforms ####################
    # Platform affix = classic_<ISA>_<µARCH>
    # Help at https://modmyclassic.com/comp
