--- a/Makefile.libretro	2021-07-29 16:25:34.978702817 +0200
+++ b/Makefile.libretro	2021-07-29 16:26:05.408702443 +0200
@@ -117,6 +117,8 @@
          PLATFORM_DEFINES += -DARM11 -marm -march=armv6j -mfpu=vfp -mfloat-abi=hard
       else ifneq (,$(findstring rpi2,$(platform)))
          PLATFORM_DEFINES += -DARM -marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+      else ifneq (,$(findstring rpi3_64,$(platform)))
+         PLATFORM_DEFINES += -DARM -mcpu=cortex-a53 -mtune=cortex-a53
       else ifneq (,$(findstring rpi3,$(platform)))
          PLATFORM_DEFINES += -DARM -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
       else ifneq (,$(findstring rpi4_64,$(platform)))
