--- a/make.sh	2021-10-12 09:38:40.824878323 +0200
+++ b/make.sh	2021-10-12 09:35:31.747103970 +0200
@@ -42,11 +42,6 @@
 OBJ_ARM64=aarch64-linux-gnu-objdump
 GCC_ARM32=arm-linux-gnueabihf-
 GCC_ARM64=aarch64-linux-gnu-
-TOOLCHAIN_ARM32=../prebuilts/gcc/linux-x86/arm/gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf/bin
-TOOLCHAIN_ARM64=../prebuilts/gcc/linux-x86/aarch64/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin
-TOOLCHAIN_ARM32_OPT=/opt/toolchains/gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf/bin
-TOOLCHAIN_ARM64_OPT=/opt/toolchains/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin
-TOOLCHAIN_ARM64=../host-toolchain-optional-linaro-aarch64-7.5-2019.12/bin
 
 ########################################### User not touch #############################################
 BIN_PATH_FIXUP="--replace tools/rk_tools/ ./"
@@ -219,20 +214,9 @@
 	local absolute_path
 
 	if grep  -q '^CONFIG_ARM64=y' ${OUTDIR}/.config ; then
-		if [ -d ${TOOLCHAIN_ARM64} ]; then
-			absolute_path=$(cd `dirname ${TOOLCHAIN_ARM64}`; pwd)
-			TOOLCHAIN_GCC=${absolute_path}/bin/${GCC_ARM64}
-			TOOLCHAIN_OBJDUMP=${absolute_path}/bin/${OBJ_ARM64}
-			TOOLCHAIN_ADDR2LINE=${absolute_path}/bin/${ADDR2LINE_ARM64}
-		elif [ -d ${TOOLCHAIN_ARM64_OPT} ]; then
-			absolute_path=$(cd `dirname ${TOOLCHAIN_ARM64_OPT}`; pwd)
-			TOOLCHAIN_GCC=${absolute_path}/bin/${GCC_ARM64}
-			TOOLCHAIN_OBJDUMP=${absolute_path}/bin/${OBJ_ARM64}
-			TOOLCHAIN_ADDR2LINE=${absolute_path}/bin/${ADDR2LINE_ARM64}
-		else
-			echo "Can't find toolchain: ${TOOLCHAIN_ARM64} or ${TOOLCHAIN_ARM64_OPT}"
-			exit 1
-		fi
+		TOOLCHAIN_GCC=${GCC_ARM64}
+		TOOLCHAIN_OBJDUMP=${OBJ_ARM64}
+		TOOLCHAIN_ADDR2LINE=${ADDR2LINE_ARM64}
 	else
 		if [ -d ${TOOLCHAIN_ARM32} ]; then
 			absolute_path=$(cd `dirname ${TOOLCHAIN_ARM32}`; pwd)
