--- a/board/rockchip/odroidgo2/odroidgo2.c	2021-08-08 16:56:51.239529580 +0200
+++ b/board/rockchip/odroidgo2/odroidgo2.c	2021-08-08 16:57:00.889529439 +0200
@@ -85,7 +85,7 @@
 	/* Enable early UART2 */
 	rk_clrsetreg(&grf->iofunc_con0,
 		     CON_IOMUX_UART2SEL_MASK,
-		     CON_IOMUX_UART2SEL_M1 << CON_IOMUX_UART2SEL_SHIFT);
+		     CON_IOMUX_UART2SEL_USBPHY << CON_IOMUX_UART2SEL_SHIFT);
 }
 
 void board_init_sfc_if(void)
