--- a/Makefile	2021-08-08 19:56:09.629370192 +0200
+++ b/Makefile	2021-08-08 19:57:01.996036083 +0200
@@ -287,6 +287,19 @@
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
    
+# Rockchip RK3399
+else ifeq ($(platform), RK3399)
+   EXT ?= so
+   CPUFLAGS += -mcpu=cortex-a72.cortex-a53 -mtune=cortex-a72.cortex-a53
+   HAVE_NEON = 1
+   TARGET := $(TARGET_NAME)_libretro.$(EXT)
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+   LIBS += -lpthread -lGLESv2
+   HAVE_OPENGLES3 = 1
+   HAVE_THREADS = 1
+   JIT_ARCH = aarch64
+   
 # Orange Pi Zero 2
 else ifeq ($(platform), orangepizero2)
    EXT ?= so
