--- a/scripts/src/3rdparty.lua	2022-08-08 11:13:57.770937867 +0200
+++ b/scripts/src/3rdparty.lua	2022-08-09 10:04:29.103342700 +0200
@@ -2458,14 +2458,11 @@
 		MAME_DIR .. "3rdparty/switchres/display_linux.h",
 		MAME_DIR .. "3rdparty/switchres/display_sdl2.cpp",
 		MAME_DIR .. "3rdparty/switchres/display_sdl2.h",
-		MAME_DIR .. "3rdparty/switchres/custom_video_xrandr.cpp",
-		MAME_DIR .. "3rdparty/switchres/custom_video_xrandr.h",
 		MAME_DIR .. "3rdparty/switchres/custom_video_drmkms.cpp",
 		MAME_DIR .. "3rdparty/switchres/custom_video_drmkms.h",
 	}
 
 	defines {
-		"SR_WITH_XRANDR",
 		"SR_WITH_KMSDRM",
 		"SR_WITH_SDL2",
 	}
@@ -2475,10 +2472,6 @@
 		backtick("pkg-config --cflags sdl2"),
 	}
 
-	links {
-		"Xrandr",
-	}
-
 	local str = backtick("pkg-config --libs libdrm")
 	addlibfromstring(str)
 	addoptionsfromstring(str)
