--- a/common/Pcsx2Defs.h	2021-11-24 00:14:03.000000000 +0100
+++ b/common/Pcsx2Defs.h	2021-11-24 01:27:47.025825126 +0100
@@ -119,7 +119,7 @@
 	#define __vectorcall __fastcall
 	#define _inline __inline__ __attribute__((unused))
 	#ifdef NDEBUG
-		#define __forceinline __attribute__((always_inline, unused))
+		#define __forceinline __attribute__((unused))
 	#else
 		#define __forceinline __attribute__((unused))
 	#endif
