--- a/cmake/BuildParameters.cmake	2021-11-24 00:14:03.000000000 +0100
+++ b/cmake/BuildParameters.cmake	2021-11-24 01:30:31.945246671 +0100
@@ -223,7 +223,7 @@
 if(MSVC)
 	add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/Zc:externConstexpr>")
 else()
-	add_compile_options(-pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse -fno-operator-names)
+	add_compile_options(-pipe -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse -fno-operator-names)
 endif()
 
 if(WIN32)
