--- a/3rdparty/CMakeLists.txt	2021-06-30 21:14:13.000000000 +0200
+++ a/3rdparty/CMakeLists.txt	2021-08-27 22:04:43.009973861 +0200
@@ -224,14 +224,12 @@
 if(USE_FAUDIO)
 	# FAudio depends on SDL2
 	find_package(SDL2)
-	if (NOT SDL2_FOUND OR SDL2_VERSION VERSION_LESS 2.0.9)
-		message("-- RPCS3: FAudio requires SDL 2.0.9 or newer.")
-	else()
-		set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared library")
-		add_subdirectory(FAudio EXCLUDE_FROM_ALL)
-		target_compile_definitions(FAudio INTERFACE -DHAVE_FAUDIO)
-		set(FAUDIO_TARGET FAudio)
-	endif()
+	message(STATUS "RPCS3: Using system FAudio")
+	find_package(FAudio REQUIRED CONFIGS FAudioConfig.cmake FAudio-config.cmake)
+	add_library(3rdparty_FAudio INTERFACE)
+	target_link_libraries(3rdparty_FAudio INTERFACE FAudio)
+	target_compile_definitions(3rdparty_FAudio INTERFACE -DHAVE_FAUDIO)
+	set(FAUDIO_TARGET 3rdparty_FAudio)
 endif()
 
 # FFMPEG
