--- a/src/types.h	2021-07-10 01:05:30.860883944 +0200
+++ b/src/types.h	2021-07-10 01:05:48.845627312 +0200
@@ -24,16 +24,8 @@
 #define STB_VORBIS_HEADER_ONLY
 #include "stb_vorbis.c"
 
-//#if !defined(_MSC_VER)
-//# include <SDL2/SDL.h>
-//# include <SDL2/SDL_image.h>
-//#else
-// These headers for SDL seem to be the pkgconfig/meson standard as per the
-// latest versions. If the old ones should be used, the ifdef must be used
-// to compare versions. 
-# include <SDL.h>
-# include <SDL_image.h>
-//#endif
+# include <SDL2/SDL.h>
+# include <SDL2/SDL_image.h>
 
 #if SDL_BYTEORDER != SDL_LIL_ENDIAN
 #error This program is not (yet) prepared for big endian CPUs, please contact the author.
