--- a/src/osd/modules/render/drawbgfx.cpp	2021-09-07 19:11:51.327168638 +0200
+++ b/src/osd/modules/render/drawbgfx.cpp	2021-09-07 20:41:46.705157509 +0200
@@ -171,7 +171,14 @@
 	}
 
 #   if BX_PLATFORM_LINUX || BX_PLATFORM_BSD || BX_PLATFORM_RPI
+#ifdef SDLMAME_NO_X11
+	if (wmi.subsystem == SDL_SYSWM_WAYLAND)
+		return (void*)wmi.info.wl.egl_window;
+       else
+		return nullptr;
+#else
 	return (void*)wmi.info.x11.window;
+#endif
 #   elif BX_PLATFORM_OSX
 	return wmi.info.cocoa.window;
 #   elif BX_PLATFORM_WINDOWS
@@ -196,8 +196,21 @@
 
 	bgfx::PlatformData pd;
 #   if BX_PLATFORM_LINUX || BX_PLATFORM_BSD
+#ifdef SDLMAME_NO_X11
+	if (wmi.subsystem == SDL_SYSWM_WAYLAND)
+	{
+		pd.ndt  = wmi.info.wl.display;
+		pd.nwh  = wmi.info.wl.egl_window;
+	}
+	else
+	{
+		pd.ndt  = NULL;
+		pd.nwh  = NULL;
+	}
+#else
 	pd.ndt          = wmi.info.x11.display;
 	pd.nwh          = (void*)(uintptr_t)wmi.info.x11.window;
+#endif
 #   elif BX_PLATFORM_OSX
 	pd.ndt          = NULL;
 	pd.nwh          = wmi.info.cocoa.window;
