--- a/scripts/src/3rdparty.lua	2022-06-21 11:18:09.152053569 +0200
+++ b/scripts/src/3rdparty.lua2	2022-06-21 12:25:04.261526856 +0200
@@ -2451,14 +2451,11 @@
 		MAME_DIR .. "3rdparty/switchres/display_linux.h",
 		MAME_DIR .. "3rdparty/switchres/display_sdl2.cpp",
 		MAME_DIR .. "3rdparty/switchres/display_sdl2.h",
-		MAME_DIR .. "3rdparty/switchres/custom_video_xrandr.cpp",
-		MAME_DIR .. "3rdparty/switchres/custom_video_xrandr.h",
 		MAME_DIR .. "3rdparty/switchres/custom_video_drmkms.cpp",
 		MAME_DIR .. "3rdparty/switchres/custom_video_drmkms.h",
 	}
 
 	defines {
-		"SR_WITH_XRANDR",
 		"SR_WITH_KMSDRM",
 		"SR_WITH_SDL2",
 	}
