--- a/CMakeLists.txt	2022-05-12 01:14:06.000000000 +0200
+++ b/CMakeLists.txt	2022-05-12 14:10:36.874851523 +0200
@@ -206,7 +206,7 @@
     unset(FN_FORCE_REQUIRED)
 endmacro()
 
-find_package(Boost 1.73.0 COMPONENTS context headers)
+find_package(Boost 1.73.0 COMPONENTS context)
 if (Boost_FOUND)
     set(Boost_LIBRARIES Boost::boost)
     # Conditionally add Boost::context only if the active version of the Conan or system Boost package provides it
